### ISSUE TITLE
Pocket error

### ISSUE THREAD

**Issue by deinemudda32** [FPtje/DarkRP#2351]
[ERROR] lua/includes/extensions/entity.lua:396: attempt to call a nil value
  1. RestoreNetworkVars - lua/includes/extensions/entity.lua:396
   2. Load - lua/includes/modules/duplicator.lua:286
    3. DoGeneric - lua/includes/modules/duplicator.lua:942
     4. deserialize - gamemodes/darkrp/entities/weapons/pocket/sv_init.lua:155
      5. dropPocketItem - gamemodes/darkrp/entities/weapons/pocket/sv_init.lua:228
       6. func - gamemodes/darkrp/entities/weapons/pocket/sv_init.lua:262
        7. unknown - lua/includes/extensions/net.lua:32

Seems to happen, when someone picks up a weapon.
Not one of my entities.

**Comment by Bo98** [comment]
> Seems to happen, when someone picks up a weapon.

Which weapon specifically? This error happens when an entity is incompatibile with GMod's own duplicator.

**Comment by deinemudda32** [comment]
M9K Weapon.

**Comment by Bo98** [comment]
I just spawned a M9K weapon through the spawn menu (Q), dropped it and I was able to pocket it fine. Is there anything particular that needs doing to cause this error?

I've seen this error a few times recently so it would be good to find why it seems to happen so often suddenly.

**Comment by Bo98** [comment]
I found the issue: old SENTs (and SWEPs) which used ent:DTVar broke while those using the newer ent:NetworkVar did not. This is a regression that arose in the December 2015 GMod update by this commit: https://github.com/garrynewman/garrysmod/commit/3e138636eb1b0ad6ed785dedf350020755cff5f1. Those using ent:DTVar did not have ent:Set[Varname] created so the error would occur for them.

I got Robotboy to fix it:

https://github.com/garrynewman/garrysmod/commit/6c31adf04e5e79180e90817d9e020af403656851

(tip: don't send pull requests)
